server/controllers/refer/courseParticipationDetailsController.ts

```diff
@@ -29,6 +29,9 @@ export default class CourseParticipationDetailsController {
         throw createError(404, `Person with prison number ${referral.prisonNumber} not found.`)
       }
 
+      FormUtils.setFieldErrors(req, res, ['yearCompleted', 'yearStarted'])
```

Comment:
Sounds like we might need to add a validation so that if anything else is added, `source` is required. Could do this in a separate PR if we need more clarity/confirmation on that